release-process.md

```diff
@@ -0,0 +1,106 @@
+# Release process
+
+This document describes our release process for open-source projects.
+
+## Contributors
+
+### New features, changes, and deprecations
+
+When adding new features (i.e., functions or classes), docstrings should properly
+indicate in which version the feature first appeared using the `.. versionadded::`
+directive.
+
+When changing the API,` .. versionchanged::` should be used. And deprecations should
+be indicated with `.. deprecated::`. Read
+[ploomber-core docs](https://ploomber-core.readthedocs.io/en/latest/deprecation.html)
+for details.
+
+
+### CHANGELOG
+
+Each repository contains a `CHANGELOG` file in the root directory. Each PR should
+contain a list of items, so we keep it up-to-date. Note that the `CHANGELOG` targets
+end-users (while `git log` targets the Ploomber development team); this implies
+that there might be changes that we don't include in the `CHANGELOG`, but they exist
+in the `git log`, for example changes to the CI configuration, new tests
+added/fixed.
+
+These are changes that we add to the `CHANGELOG`:
+
+- [Doc] Important documentation changes
+- [Feature] New features
+- [API Change] API breaking changes
+- [Fix] Bug fixes
+
+Each new line in the `CHANGELOG` must be prefixed by its category. Example:
+
+```md
+- [Fix] Fixes an error that caused function `do_something` to break when passing `0` as input
+```
+
+If there is an issue related to the change, it should be added to the end:
+
+```md
+- [Fix] Fixes an error that caused function `do_something` to break when passing `0` as input (#99)
+```
+
+Note that we're not adding the link to GitHub, this will happen automatically during the release process.
+
+## Reviewing Pull Requests
+
+Once the Pull Request is reviewed and approved, the reviewer determines
+what happens next. Most often, the following action would be to deploy a new minor version
+to PyPI, so once the PR is merged and the CI passes in `master`/`main`, a new minor version will be uploaded to PyPI.
+
+Another action can be "no release", implying that the PR is merged, but no release
```

Comment:
How does the reviewer pass these flags?

Reply:
Also, shouldn't we make it automatic?
This adds complexity to the review process which is already pretty time consuming.

Reply:
> How does the reviewer pass these flags?

to the commit message when merging

> Also, shouldn't we make it automatic?

This is challenging to automate. How can a script determine if this is a minor or major release? It'd have to analyze the API and look for breaking changes. This also adds accountability since the reviewer is responsible for signing off.

Reply:
just a clarifying note: I'm not against having some automation here. but I think this has to be a human decision. however, we can implement it with software. e.g., write a program that checks for API breaking changes so the reviewer can quickly determine if this will be a major release (opened an issue here https://github.com/ploomber/pkgmt/issues/30); however, I think we shouldn't wait to have that program ready to get started with this

Reply:
Sounds good ^^
As long as it's not blocking I'm fine. I do think for the major/minor you're right, maybe this script can be a check in the CI so the reviewer knows there's a warning about backward compatibility.

Reply:
ok, let's leave the issue open on pkgmt  and for now, the reviewer must sign off the release